File: app/src/main/java/com/example/wangbin/gymclub/adapter/CourseAdapter.java
package com.example.wangbin.gymclub.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.wangbin.gymclub.R;
import com.example.wangbin.gymclub.VideoActivity;

import java.util.List;

public class CourseAdapter extends RecyclerView.Adapter<CourseAdapter.CourseHolder>{
    private Context context;
    private List<HolderItem> holderItems;
    public CourseAdapter(Context context, List<HolderItem> holderItems){
        this.context = context;
        this.holderItems = holderItems;
    }
    @NonNull
    @Override
    public CourseHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.course_item,viewGroup,false);
        return new CourseHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CourseHolder courseHolder, int i) {
        courseHolder.bind(holderItems.get(i));
    }

    @Override
    public int getItemCount() {
        return holderItems.size();
    }

    class CourseHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        private TextView textView;
        private ImageView imageView;

        public CourseHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.tv_course);
            imageView = itemView.findViewById(R.id.img_course);
            itemView.setOnClickListener(this);
        }

        public void bind(HolderItem holderItem){
            textView.setText(holderItem.text);
            imageView.setImageResource(holderItem.img_id);
        }

        @Override
        public void onClick(View v) {
            Intent i = new Intent(context, VideoActivity.class);
            context.startActivity(i);
        }
    }
    public static class HolderItem{
        public String text;
        public int img_id;

        public HolderItem(String text, int img_id) {
            this.text = text;
            this.img_id = img_id;
        }
    }
}

File: app/src/main/java/com/example/wangbin/gymclub/adapter/CourseAdapter.kt
package com.example.wangbin.gymclub.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.example.wangbin.gymclub.R
import kotlinx.android.synthetic.main.course_item.view.*
class CourseAdapter : RecyclerView.Adapter<CourseAdapter.CourseHolder> {
    private var mContext:Context? = null
    private var mData:List<HolderItem>? = null
    constructor(context:Context,data:List<HolderItem>):super(){
        this.mContext = context
        this.mData = data
    }
    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): CourseHolder {
        var courseHolder = CourseHolder(LayoutInflater.from(mContext).inflate(R.layout.course_item,p0,false))
        return courseHolder
    }

    override fun getItemCount(): Int {
       return mData?.size!!
    }

    override fun onBindViewHolder(p0: CourseHolder, p1: Int) {
       p0.bind(mData?.get(p1)!!)
    }

    class CourseHolder : RecyclerView.ViewHolder {
        private var mTvCourse : TextView? = null
        private var mImgCourse : ImageView? = null
        constructor(view: View):super(view){
            mTvCourse = view.findViewById(R.id.tv_course)
            mImgCourse = view.findViewById(R.id.img_course)
        }

        fun bind(holderItem: HolderItem){
            mTvCourse?.text = holderItem.text
            mImgCourse?.setImageResource(holderItem.img_url_id)
        }

    }

    class HolderItem(val text: String,val img_url_id: Int)
}